cmd/apply/cmdapply.go

```diff
@@ -58,6 +58,10 @@ func GetApplyRunner(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *A
 		"If true, do not prune previously applied objects.")
 	cmd.Flags().StringVar(&r.prunePropagationPolicy, "prune-propagation-policy",
 		"Background", "Propagation policy for pruning")
+	cmd.Flags().BoolVar(&r.waitForPrune, "wait-for-prune", false,
+		"Wait for all pruned resources to be deleted.")
+	cmd.Flags().DurationVar(&r.waitForPruneTimeout, "wait-for-prune-timeout", 1*time.Minute,
```

Comment:
Do we need 2 flags here ? If user specifies waitForPruneTimeout, shouldn't be good enough? Adding 2 flags might lead to confusion. What if waitForPruneTimeout is specified and waitForPrune is not.

Reply:
That is an interesting suggestion. If so, we should update all the `--wait-..` flags to follow the same convention.

Reply:
So a solution where we only use one flag per operation, we would end up with something like:
`--wait-for-reconcile=2m`
`--wait-for-prune=1m`
It would no longer be valid to use just `--wait-for-<something>`, the user will need to provide a timeout value. This isn't too bad I think, choosing default values is difficult since it depends so much on the type of resources being applied/pruned.

Any thoughts on this? @seans3 @monopole

Reply:
One flag per wait operation (instead of two) seems simpler. I can't think of any drawbacks. I think it's a good idea.

Reply:
And I would rename the flags to be 

`--reconcile-timeout=2m`
`--prune-timeout=1m`

Reply:
Yeah, I agree. I would prefer if we can merge this as is, and I will do a follow-up PR to fix the flags across all.